File: frontend/src/api/aiMaterial.ts
import api from './client'
import type { Song } from '../types'

export interface MaterialExpandResult {
  query: string
  intent: {
    raw_text: string
    query: string
    artist?: string | null
    moods: string[]
    scenes: string[]
    energy: 'low' | 'medium' | 'high'
  }
  found: number
  added: number
  total_material: number
  songs: Song[]
  message: string
  error?: string
}

export async function expandMaterial(text: string, limit = 12) {
  const { data } = await api.post('/ai-material/expand', { text, limit })
  return data as MaterialExpandResult
}

File: backend/app/routers/ai_material.py
from fastapi import APIRouter, Depends, Request
from pydantic import BaseModel
from sqlalchemy.ext.asyncio import AsyncSession

from ..database import get_session
from ..services.public_library_service import expand_ai_material

router = APIRouter(prefix="/api/ai-material", tags=["ai-material"])


class ExpandRequest(BaseModel):
    text: str
    limit: int = 12


@router.post("/expand")
async def expand_material(body: ExpandRequest, request: Request, session: AsyncSession = Depends(get_session)):
    user_id = getattr(request.state, "user_id", None)
    if not user_id:
        return {"error": "Missing client identity", "found": 0, "added": 0, "songs": []}
    text = body.text.strip()
    if not text:
        return {"error": "请输入想找的歌、艺人或场景", "found": 0, "added": 0, "songs": []}
    return await expand_ai_material(session, user_id, text, min(max(body.limit, 1), 30))
